ingestion_service/producer/core/routers/api_endpoints_input.py

```diff
@@ -0,0 +1,159 @@
+                                      "user": "U123r",
+                                      "type": "json",
+                                      "date_created": "2024-04-30T12:42:32.203447",
+                                      "date_modified": "2024-04-30T12:42:32.203451",
+                                      "json_data": {
+
+                                          "neuroscience_disorders": [
+                                              {
+                                                  "disorder": "Alzheimer's disease",
+                                                  "description": "A progressive neurodegenerative disorder that leads to memory loss and cognitive decline.",
+                                                  "symptoms": ["Memory loss", "Confusion",
+                                                               "Trouble with language and reasoning"],
+                                                  "treatments": ["Medications (e.g., cholinesterase inhibitors)",
+                                                                 "Cognitive therapy"]
+                                              },
+                                              {
+                                                  "disorder": "Parkinson's disease",
+                                                  "description": "A neurodegenerative disorder characterized by tremors, rigidity, and difficulty with movement.",
+                                                  "symptoms": ["Tremors", "Bradykinesia", "Postural instability"],
+                                                  "treatments": ["Levodopa", "Deep brain stimulation (DBS)"]
+                                              },
+                                              {
+                                                  "disorder": "Schizophrenia",
+                                                  "description": "A chronic and severe mental disorder that affects how a person thinks, feels, and behaves.",
+                                                  "symptoms": ["Hallucinations", "Delusions", "Disorganized thinking"],
+                                                  "treatments": ["Antipsychotic medications", "Psychotherapy"]
+                                              }
+                                          ]
+
+                                      }
+                                  }
+                              ],
+                          ),
+                      ], ):
+    try:
+        main_model_schema = jsoninput.json()
+        publish_message(main_model_schema)
+    except json.JSONDecodeError as e:
+        raise HTTPException(status_code=400, detail="Invalid JSON" + str(e))
+
+    return JSONResponse(content={"message": f"Data uploaded successfully {main_model_schema}"})
+
+
+@router.post("/ingest/raw/jsonld")
+async def ingest_json(user: Annotated[LoginUserIn, Depends(get_current_user)],
```

Comment:
There is no `await` call here -- is this truly `async`?

Reply:
I will check but updating the code to make async, I think it would be better to do in the second version after all the required features have been implemented, a improvement. Let me know your thoughts on it?

Reply:
I'd remove the `async` then on the function call if it isn't truly `async`

Reply:
It's the standard fastapi approach --> [https://fastapi.tiangolo.com/tutorial/first-steps/#interactive-api-docs](https://fastapi.tiangolo.com/tutorial/first-steps/#interactive-api-docs)